docs/types.rst

```diff
@@ -0,0 +1,318 @@
+
+
+Runtime Properties
+==================
+
+See section on `runtime properties <http://cloudify-plugins-common.readthedocs.org/en/3.3/context.html?highlight=runtime#cloudify.context.NodeInstanceContext.runtime_properties>`_
+
+Node instances of any of the types defined in this plugin
+get set with the following runtime properties
+during the ``cloudify.interfaces.lifecycle.create`` operation:
+
+* ``aws_resource_id`` the AWS ID of the resource
+
+
+Using Existing Resources
+========================
+
+It is possible to use existing resources on AWS - whether these have been created by a different Cloudify deployment or not via Cloudify at all.
+
+All Cloudify AWS types have a property named ``use_external_resource``, whose default value is ``false``. When set to ``true``, the plugin will apply different semantics for each of the operations executed on the relevant node's instances:
+
+.. note::
+    If ``use_external_resource`` is set to true in the blueprint,
+    the ``resource_id`` must be that resource's ID in AWS,
+    unless the resource type is a keypair,
+    in which case the value is the key's name.
+
+This behavior is common to all resource types:
+
+* ``create`` If ``use_external_resource`` is true, the AWS plugin will check if the resource is available in your account. If no such resource is available, the operation will fail, if it is available, it will assign the ``aws_resource_id`` to the instance ``runtime_properties``.
+* ``delete`` If ``use_external_resource`` is true, the AWS plugin will check if the resource is available in your account. If no such resource is available, the operation will fail, if it is available, it will unassign the instance ``runtime_properties``.
+
+
+Relationships
+=============
+
+.. cfy:rel:: cloudify.aws.relationships.instance_connected_to_keypair
+
+   The `run_instances` operation looks to see if there are any relationships that define a relationship between the instance and a keypair.
+   If so, that keypair will be the keypair for that instance.
+   It inserts the key's name property in the ``key_name`` parameter in the `run_instances` function.
+
+
+.. cfy:rel:: cloudify.aws.relationships.instance_connected_to_security_group
+
+   The `run_instances` operation looks to see if there are any relationships that define a relationship between the instance and a security group.
+   If so, that security group's ID will be the included in the list of security groups in the ``security_group_ids`` parameter in the `run_instances` function.
+
+
+.. cfy:rel:: cloudify.aws.relationships.instance_connected_to_subnet
+
+   The `run_instances` operation looks for any relationships to
+   a Subnet and creates the Instance in that Subnet.
+   Otherwise, the instance is in the EC2 Classic VPC.
+
+.. cfy:rel:: cloudify.aws.relationships.connected_to_subnet
+
+   TODO: WAT
+
+
+.. cfy:rel:: cloudify.aws.relationships.instance_contained_in_subnet
+.. warning::
+   Deprecated! Please use
```

Comment:
I have just now decided that this relationship is deprecated. Please shout if I'm wrong (but if I'm wrong we should probably have a discussion about it)